### DISCUSSION TITLE
How to convert a dynamic shovel to a static using the advanced.config?

### DISCUSSION THREAD

**Discussion by sdancer75** [rabbitmq/rabbitmq-server#10318] [Other] [answered]
How to convert the dynamic shovel shown in the screenshot into a dynamic one?

![shovel_local_to_cloud](https://github.com/rabbitmq/rabbitmq-server/assets/10809692/2c15e3db-d377-46aa-9ac3-3363f8650b13)

So far I tried this but it is terminated immediately.

```

{rabbitmq_shovel,
  [ {shovels, [ 
  
  

            {from_local_to_cloud, [  
               {source, [
                 {protocol, amqp091},
                 {uris, ["amqp://"]},                    
                 {declarations, [
                                    
                 ]},
                 {queue, <<"local_to_cloud">>}
                ]},

            
               {destination, [
                 {protocol, amqp091},
                 {uris, ["amqp://dzjkevdt:<password>@sparrow.rmq.cloudamqp.com/dzjkevdt"]},                  
                 {declarations, [
                        
                 ]},
                 {queue, <<"local_to_cloud">>}
                ]}      
                
            ]
        }             
                  
                  
                  
     ]}
  ]},


```

**Comment by michaelklishin** [comment]
You have a comma at the end of this config while it must be a dot (if this is all you have in `advanced.config`, that is).

**Comment by michaelklishin** [comment]
When to command is replaced with a dot, the contents read as syntactically valid Erlang term. The structure seems reasonable.

See node logs for clues. We are very bad at mind reading.

**Reply by sdancer75** [reply, marked as answer]
Hey, thanks for your answer, the code snippet in my post it was just a part of the whole `advanced.config` file.


Anyway, I had to manually create the queue first. I thought it would be created automatically (as in creating dynamic shovels), but this was not true.

Now the static configuration into the `advanced.config` file for shovel's looks like this and they seem to work fine.

```
{from_local_to_cloud, 
        [  
           {source, [
             {protocol, amqp091},
             {uris, ["amqp://"]},                    
             {declarations, []},
             {queue, <<"local_to_cloud">>}                 
            ]},

        
           {destination, [
             {protocol, amqp091},
             {uris, ["amqp://dzjkevdt:<password>@sparrow.rmq.cloudamqp.com/dzjkevdt"]},                  
             {declarations, []},
             {queue, <<"local_to_cloud">>}
            ]}
            
        ]
    }
    
```

**Reply by mkuratczyk** [reply]
There's a `declarations` section in the configuration to declare queues if you want.
